polaris-core/src/main/java/org/apache/polaris/core/identity/dpo/ServiceIdentityInfoDpo.java

```diff
@@ -72,10 +74,25 @@ public SecretReference getIdentityInfoReference() {
   }
 
   /**
-   * Converts this persistence object to the corresponding API model. During the conversion, some
-   * fields will be dropped, e.g. the reference to the service identity's credential
+   * Converts this persistence object to the corresponding API model.
+   *
+   * <p>The conversion uses the provided {@link ServiceIdentityProvider} to retrieve the user-facing
+   * identity information (e.g., AWS IAM ARN) without exposing sensitive credentials. The credential
+   * reference stored in this DPO is not included in the API model.
+   *
+   * @param serviceIdentityProvider the service identity provider used to retrieve display
+   *     information
+   * @return the API model representation, or null if the provider is null or cannot resolve the
+   *     identity
    */
-  public abstract @Nonnull ServiceIdentityInfo asServiceIdentityInfoModel();
+  public @Nullable ServiceIdentityInfo asServiceIdentityInfoModel(
+      ServiceIdentityProvider serviceIdentityProvider) {
+    if (serviceIdentityProvider == null) {
```

Comment:
if `serviceIdentityProvider` is `null`, why would the caller even call this method?

Currently `serviceIdentityProvider` is ultimately injected, so it should not be `null`, right?

Reply:
Done!